ComposedTests/Complex+Spec.swift

```diff
@@ -0,0 +1,231 @@
+                }
+            }
+
+            context("when segmented.selectedChild == nil") {
+                beforeEach {
+                    segmented.setSelected(index: nil)
+                }
+
+                it("section counts should be as expected") {
+                    expect(segmented.numberOfSections).to(equal(0))
+                    expect(innerComposed.numberOfSections).to(equal(1))
+                    expect(outerComposed.numberOfSections).to(equal(5))
+                }
+
+                it("local section data should be valid") {
+                    let actual0 = outerComposed.localSection(for: 0)
+                    let actual1 = outerComposed.localSection(for: 1)
+                    let actual2 = outerComposed.localSection(for: 2)
+                    let actual3 = outerComposed.localSection(for: 3)
+                    let actual4 = outerComposed.localSection(for: 4)
+
+                    expect(DataSourceHashableWrapper(actual0.dataSource)).to(equal(DataSourceHashableWrapper(sectioned)))
+                    expect(DataSourceHashableWrapper(actual1.dataSource)).to(equal(DataSourceHashableWrapper(sectioned)))
+                    expect(DataSourceHashableWrapper(actual2.dataSource)).to(equal(DataSourceHashableWrapper(list3)))
+                    expect(DataSourceHashableWrapper(actual3.dataSource)).to(equal(DataSourceHashableWrapper(countries)))
+                    expect(DataSourceHashableWrapper(actual4.dataSource)).to(equal(DataSourceHashableWrapper(empty2)))
+                }
+            }
+
+            context("when segmented is removed from innerComposed") {
+                beforeEach {
+                    innerComposed.remove(dataSource: segmented)
+                }
+
+                it("section counts should be as expected") {
+                    expect(innerComposed.numberOfSections).to(equal(1))
+                    expect(outerComposed.numberOfSections).to(equal(5))
+                }
+
+                it("local section data should be valid") {
+                    let actual0 = outerComposed.localSection(for: 0)
+                    let actual1 = outerComposed.localSection(for: 1)
+                    let actual2 = outerComposed.localSection(for: 2)
+                    let actual3 = outerComposed.localSection(for: 3)
+                    let actual4 = outerComposed.localSection(for: 4)
+
+                    expect(DataSourceHashableWrapper(actual0.dataSource)).to(equal(DataSourceHashableWrapper(sectioned)))
+                    expect(DataSourceHashableWrapper(actual1.dataSource)).to(equal(DataSourceHashableWrapper(sectioned)))
+                    expect(DataSourceHashableWrapper(actual2.dataSource)).to(equal(DataSourceHashableWrapper(list3)))
+                    expect(DataSourceHashableWrapper(actual3.dataSource)).to(equal(DataSourceHashableWrapper(countries)))
+                    expect(DataSourceHashableWrapper(actual4.dataSource)).to(equal(DataSourceHashableWrapper(empty2)))
```

Comment:
It seems odd to me that I get back a wrapper from these. Say if I wanted to get the data source and cast it to the data source that I know I inserted the cast would fail?